scripts/simulateFunctionsJavaScript.js

```diff
@@ -1,12 +1,37 @@
-const { simulateRequest, getDecodedResultLog } = require("../FunctionsSandboxLibrary")
-const requestConfig = require("../Functions-request-config.js")
+const { simulateRequest, getDecodedResultLog, getRequestConfig } = require("../FunctionsSandboxLibrary")
 
-;(async () => {
+const runSimulation = async (requestConfig) => {
   const { resultLog, result, success } = await simulateRequest(requestConfig)
 
   console.log(`\n${resultLog}`)
   if (success) {
-    console.log(`Value returned from souce code: ${result}\n${getDecodedResultLog(requestConfig, result)}`)
+    console.log(`Value returned from source code: ${result}\n${getDecodedResultLog(requestConfig, result)}`)
     return
   }
+}
+
+(async () => {
+    const unvalidatedRequestConfig = require("../Functions-request-config.js")
+    const requestConfig = getRequestConfig(unvalidatedRequestConfig)
+
+    if (requestConfig.secretsLocation === 1) {
+      if (requestConfig.secrets && Object.keys(requestConfig.secrets).length !== 0) {
+        console.log('\n__SIMULATING JAVASCRIPT WITH DEFAULT SECRETS__'.toUpperCase())
+        await runSimulation(requestConfig)
+      }
+
+      if (!requestConfig.perNodeSecrets) {
+        return
+      }
+
+      for (let i = 0; i < requestConfig.perNodeSecrets.length; i++) {
```

Comment:
I feel like this should be under a flag. Not everybody wants to repeat a simulation N times every single time.

Reply:
The problem is there is no consistent cross-platform way to set flags for `npm` commands.

Maybe I'll just let a user set an env var if they want to turn this off.